drivers/client_location_history/app/controllers/client_location_history/warehouse_reports/client_location_history_controller.rb

```diff
@@ -15,7 +15,7 @@ class ClientLocationHistoryController < ApplicationController
 
     def index
       ids = ClientLocationHistory::Location.joins(:client).
-        merge(GrdaWarehouse::Hud::Client.destination_visible_to(current_user)).
+        merge(GrdaWarehouse::Hud::Client.destination_or_source_visible_to(current_user)).
```

Comment:
Again, you probably want to hint here as well.
```suggestion
        merge(GrdaWarehouse::Hud::Client.destination_or_source_visible_to(current_user, source_client_ids: @client.source_client_ids)).
```

Reply:
This section is looking for any/all clients that match so we want to bring all of the possible clients back here.